app/src/util/tools.ts

```diff
@@ -45,9 +45,15 @@ export const mulBN = (a: BigNumber, b: number, scale = 10000): BigNumber => {
 export const calcPrice = (holdingsBN: BigNumber[]): number[] => {
   const holdings = holdingsBN.map(h => new Big(h.toString()))
   const product = holdings.reduce((a, b) => a.mul(b))
-  const denominator = holdings.map(h => product.div(h)).reduce((a, b) => a.add(b))
-
-  const prices = holdings.map(holding => product.div(holding).div(denominator))
+  const denominator = holdings
+    .map(h => (h.toString() === '0' ? new Big(0) : product.div(h)))
+    .reduce((a, b) => a.add(b))
+
+  const prices = holdings.map(holding =>
+    product.toString() === '0' || denominator.toString() === '0'
+      ? new Big(0)
+      : product.div(holding).div(denominator),
+  )
```

Comment:
I think this fix this problem https://github.com/protofire/gnosis-conditional-exchange/issues/379

Reply:
This is probably fine for now, but the approach is unnecessarily involved. I would just add a check at the beginning of the function: if any holding is 0, then return all zeroes. (In practice, I think that either *all* holdings are zero, or all are non-zero, but anyway).